### PR TITLE
fix: Address compilation errors when using the agent with Typescript

### PR DESCRIPTION
**Pull request by DominicKramer**
fixes: #367

### CODE REVIEW

File: src/agent/config.ts [hunk 1]
```diff
@@ -119,54 +119,86 @@ export interface DebugAgentConfig extends common.AuthenticationConfig {
...
-  forceNewAgent_: boolean;
+  forceNewAgent_?: boolean;
 
   /**
    * Uses by tests to cause the start() function to return the debuglet.
    */
+  testMode_?: boolean;
+}
+
+export interface ResolvedDebugAgentConfig extends DebugAgentConfig {
```
Comment (kjin) [inline src/agent/config.ts]: Do [`Partial<T>`](https://www.typescriptlang.org/docs/handbook/advanced-types.html) types work here?

**Comment by kjin** [comment]
Additional nit -- PR title should be `fix:`, not `bug:`, right?

File: src/agent/config.ts (hunk 1 above)
Comment (DominicKramer) [reply src/agent/config.ts]: Nice.  I wasn't aware of `Partial<T>` types.  I've updated the code to use them.

File: src/agent/config.ts [hunk 2]
```diff
@@ -16,7 +16,11 @@
 
 import * as common from '../types/common';
 
-export interface DebugAgentConfig extends common.AuthenticationConfig {
+export type DebugAgentConfig = {
+  [K in keyof ResolvedDebugAgentConfig]?: Partial<ResolvedDebugAgentConfig[K]>
```
Comment (kjin) [inline src/agent/config.ts]: Does `export type DebugAgentConfig = Partial<ResolvedDebugAgentConfig>` work?

Also, can you check to see that doc comments still show up in VS Code intellisense?

Comment (DominicKramer) [reply src/agent/config.ts]: `export type DebugAgentConfig = Partial<ResolvedDebugAgentConfig>` does work.  Also, VS Code intellisense still shows the docs for `DebugAgentConfig`.

**Comment by DominicKramer** [comment]
@kjin I uploaded a new commit that added additional installation tests.

**Pull request merged**